rasterio/plot.py

```diff
@@ -105,31 +105,34 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
                 # will have the colors in the order expected by
                 # matplotlib (RGB)
                 arr = source.read(rgb_indexes, masked=True)
-                arr = reshape_as_image(arr)
-
             except KeyError:
-                arr = source.read(1, masked=True)
+                if indexes == None: indexes = (0,1,2)
+                indexes = tuple(i+1 for i in indexes)
+                arr = source.read(indexes, masked=True)
+
+            arr = reshape_as_image(arr)
     else:
         # The source is a numpy array reshape it to image if it has 3+ bands
         if source.ndim >= 3:
             source = np.ma.squeeze(source)
 
         if source.ndim >= 3:
             arr = reshape_as_image(source)
+            if indexes == None: indexes = (0,1,2)
+            if arr.shape[-1]>3: arr = arr[:, :, indexes]
```

Comment:
here I'll just do something like 

```
arr = reshape_as_image(source[:, :, 0:3])
```

Reply:
we have modified the related scripts in our latest code file, please check it.